scheduler/scheduler.cpp

```diff
@@ -2245,6 +2273,24 @@ int main(int argc, char *argv[])
                 max_fd--;
             }
         }
+
+        for (list<pair<int, CompileServer *> >::const_iterator it = sock_to_cs_list.begin();
+                it != sock_to_cs_list.end(); ++it) {
+            int i = (*it).first;
+            CompileServer *cs = (*it).second;
+
+            if (max_fd && FD_ISSET(i, &read_set) || FD_ISSET(i, &write_set)) {
```

Comment:
It feels like we should just notify CS that select returned, with two bool paramaters: FD_ISSET(i, read_set), FD_ISSET(i, write_set) and let the cs figure out what to do about it.